File: Java/dependency-injection/_01_changing-behavior-through-composition/src/main/java/com/github/grzesiek_galezowski/ChangingBehaviorThroughComposition/Alarms/LoudAlarm.java
package com.github.grzesiek_galezowski.ChangingBehaviorThroughComposition.Alarms;

import com.github.grzesiek_galezowski.ChangingBehaviorThroughComposition.Alarm;

public class LoudAlarm implements Alarm {
  public void trigger() {
    System.out.println("Playing very loud sound!");
  }

  public void disable() {
    System.out.println("Not playing very loud sound anymore");
  }
}

File: CSharp/DependencyInjection/ChangingBehaviorThroughComposition/Alarms/LoudAlarm.cs
using System;

namespace ChangingBehaviorThroughComposition
{
  public class LoudAlarm : Alarm
  {
    public void Trigger()
    {
      Console.WriteLine("Playing very loud sound!");
    }

    public void Disable()
    {
      Console.WriteLine("Not playing very loud sound anymore");
    }
  }
}